openedx_external_enrollments/edxapp_wrapper/backends/course_settings_j_v1.py

```diff
@@ -0,0 +1,73 @@
+"""Backend for course settings module."""
+import logging
+import time
+
+from cms.djangoapps.models.settings.course_metadata import CourseMetadata
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx_external_enrollments.models import OtherCourseSettings
+from xmodule.modulestore.django import modulestore
+
+LOG = logging.getLogger(__name__)
+
+
+def update_course_settings(*args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Updates external_course_id and external_platform from a course overview.
+    This function is called from edx-platform once advance course settings is successfully saved.
+    """
+    try:
+        other_course_settings = kwargs.get('other_course_settings', {})
+
+        OtherCourseSettings.objects.update_or_create(
+            course_overview=CourseOverview.get_from_id(kwargs.get('course_key')),
+            defaults={
+                'external_course_id': other_course_settings.get('external_course_run_id', ''),
+                'external_platform': other_course_settings.get('external_platform_target', ''),
+            },
+        )
+    except Exception as error:  # pylint: disable=broad-except
+        LOG.error('Failed to update course_settings in the backend. Reason: %s', str(error))
+
+
+def migrate_course_settings(*args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Migrates all course settings to OtherCourseSettings table in edxapp database.
+    delay_seconds -> sleep time in seconds.
+    group_length -> number of courses to be migrated by groups.
+    """
+    migrated_courses = []
+
+    try:
+        group_counter = 0
+
+        for course in modulestore().get_courses():
+            group_counter += 1
+            course_overview = CourseOverview.get_from_id(course.id)
+            other_course_settings_key = CourseMetadata.fetch(course).get('other_course_settings', {})
+            other_course_settings = other_course_settings_key.get('value', {})
```

Comment:
```suggestion
            other_course_settings = CourseMetadata.fetch(course).get('other_course_settings', {}).get('value', {})
```
I think it is fine to chain these "getters". @ivanvgh

Reply:
Alright.